### ISSUE TITLE
40 lines is far too short for a unit test

### ISSUE THREAD

**Issue by brian-armstrong** [realm/SwiftLint#2222]
Hi,

I recently tried out SwiftLint to run against the Swift unit tests on my objc project. To my surprise, the tool complains about functions that are longer than 40 lines. I have probably seen more unit tests that *are* longer than 40 than are not. This rule shouldn't run on unit tests, it's arguably nonsensical there.

**Comment by ornithocoder** [comment]
@brian-armstrong, you can disable it for unit tests by adding a nested `.swiftlint.yml` which disables it.

```yml
  disabled_rules:
    - function_body_length
    - type_body_length
```

**Comment by brian-armstrong** [comment]
Sure, but isn't the default behavior busted?

**Comment by marcelofabri** [comment]
There's no way a default value will please everyone, that's why that rule is configurable 😉

**Comment by jpsim** [comment]
You should definitely feel free to deviate from the defaults provided by SwiftLint. I do stand by the 40 line default however, especially if you abide by the ["tests should only test one thing at a time" philosphy](https://stackoverflow.com/a/61868/373262). Many reasonable people will disagree with that mantra, but it's a good starting point IMHO.
